lib/navigation/route.dart

```diff
@@ -1,31 +1,40 @@
 import 'package:flutter/material.dart';
 import 'package:go_router/go_router.dart';
+import 'package:injectable/injectable.dart';
 import 'package:survey_flutter_ic/ui/home/home_screen.dart';
 import 'package:survey_flutter_ic/ui/signin/sign_in_screen.dart';
 import 'package:survey_flutter_ic/ui/splash/splash_screen.dart';
 
-const routePathRootScreen = '/';
-const routePathHomeScreen = '/home';
-const routePathSignInScreen = '/sign_in';
+enum RoutePath {
+  root('/'),
+  home('/home'),
+  signIn('/sign_in');
 
-// TODO Extract to class and inject by GetIt
-GoRouter router([String? initialLocation]) => GoRouter(
-      initialLocation: initialLocation ?? routePathRootScreen,
-      routes: <GoRoute>[
-        GoRoute(
-          path: routePathRootScreen,
-          builder: (BuildContext context, GoRouterState state) =>
-              const SplashScreen(),
-        ),
-        GoRoute(
-          path: routePathSignInScreen,
-          builder: (BuildContext context, GoRouterState state) =>
-              const SignInScreen(),
-        ),
-        GoRoute(
-          path: routePathHomeScreen,
-          builder: (BuildContext context, GoRouterState state) =>
-              const HomeScreen(),
-        ),
-      ],
-    );
+  const RoutePath(this.path);
+
+  final String path;
+}
+
+@Singleton()
+class AppRouter {
+  GoRouter router([String? initialLocation]) => GoRouter(
+        initialLocation: initialLocation ?? RoutePath.root.path,
+        routes: <GoRoute>[
+          GoRoute(
+            path: RoutePath.root.path,
+            builder: (BuildContext context, GoRouterState state) =>
+                const SplashScreen(),
+          ),
+          GoRoute(
+            path: RoutePath.home.path,
+            builder: (BuildContext context, GoRouterState state) =>
+                const SignInScreen(),
+          ),
+          GoRoute(
+            path: RoutePath.signIn.path,
+            builder: (BuildContext context, GoRouterState state) =>
+                const HomeScreen(),
+          ),
```

Comment:
@Wadeewee I would suggest defining `name` for each route because it would be helpful later 😉 

For example, when you use `path` to navigate to the detail screen (the `path` for detail screen can be `/survey/{id}`), you must figure out the full path for the detail screen with `context.go('/home/survey/{id}')` (Add home path to before the survey path). With `name`, we can simply call `context.goNamed('survey-detail')`

Reply:
Ah, I see 👍 Thanks for the suggestion @manh-t 🤗 I will try it out and update soon 🚀